planner/core/exhaust_physical_plans.go

```diff
@@ -450,11 +454,10 @@ func (p *LogicalJoin) getHashJoins(prop *property.PhysicalProperty) (joins []Phy
 		}
 	}
 
-	forced = (p.preferJoinType&preferHashJoin > 0) || forceLeftToBuild || forceRightToBuild
-	noHashJoin := (p.preferJoinType & preferNoHashJoin) > 0
-	if !forced && noHashJoin {
+	forced = forceLeftToBuild || forceRightToBuild
+	if !forced && p.shouldSkipHashJoin() {
 		return nil, false
-	} else if forced && noHashJoin {
+	} else if forced && p.shouldSkipHashJoin() {
 		p.SCtx().GetSessionVars().StmtCtx.AppendWarning(ErrInternal.GenWithStack(
 			"Some HASH_JOIN and NO_HASH_JOIN hints conflict, NO_HASH_JOIN is ignored"))
```

Comment:
The warning message should be updated, in the case below, the warning message is not accurate:
```
mysql> set tidb_opt_enable_hash_join=off;
Query OK, 0 rows affected (0.00 sec)

mysql> explain select /*+ hash_join_build(t1) */ * from t1, t2;
+-----------------------------+--------------+-----------+---------------+--------------------------------+
| id                          | estRows      | task      | access object | operator info                  |
+-----------------------------+--------------+-----------+---------------+--------------------------------+
| HashJoin_8                  | 100000000.00 | root      |               | CARTESIAN inner join           |
| ├─TableReader_10(Build)     | 10000.00     | root      |               | data:TableFullScan_9           |
| │ └─TableFullScan_9         | 10000.00     | cop[tikv] | table:t1      | keep order:false, stats:pseudo |
| └─TableReader_12(Probe)     | 10000.00     | root      |               | data:TableFullScan_11          |
|   └─TableFullScan_11        | 10000.00     | cop[tikv] | table:t2      | keep order:false, stats:pseudo |
+-----------------------------+--------------+-----------+---------------+--------------------------------+
5 rows in set, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+-------------------------------------------------------------------------+
| Level   | Code | Message                                                                 |
+---------+------+-------------------------------------------------------------------------+
| Warning | 1815 | Some HASH_JOIN and NO_HASH_JOIN hints conflict, NO_HASH_JOIN is ignored |
+---------+------+-------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
How did you manage to load t1, t2 data into above tidb cluster?

Reply:
Emm no data in them. I just create 2 empty tables and then use `stats:pseudo`.